paymentsheet/src/main/java/com/stripe/android/paymentsheet/elements/common/TextFieldComposable.kt

```diff
@@ -0,0 +1,91 @@
+package com.stripe.android.paymentsheet.elements
+
+import android.util.Log
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.text.KeyboardOptions
+import androidx.compose.material.LocalContentAlpha
+import androidx.compose.material.LocalContentColor
+import androidx.compose.material.MaterialTheme
+import androidx.compose.material.Text
+import androidx.compose.material.TextField
+import androidx.compose.material.TextFieldDefaults
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.livedata.observeAsState
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.saveable.rememberSaveable
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.focus.FocusRequester
+import androidx.compose.ui.focus.focusOrder
+import androidx.compose.ui.focus.isFocused
+import androidx.compose.ui.focus.onFocusChanged
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.input.ImeAction
+import com.stripe.android.paymentsheet.elements.common.TextFieldElement
+
+/** This is a helpful method for setting the next action based on the nextFocus Requester **/
+fun imeAction(nextFocusRequester: FocusRequester?): ImeAction = nextFocusRequester?.let {
+    ImeAction.Next
+} ?: ImeAction.Done
+
+/**
+ * This is focused on converting an `Element` into what is displayed in a textField.
+ * - some focus logic
+ * - observes values that impact how things show on the screen
+ * - calls through to the Elements worker functions for focus change and value change events
+ */
+@Composable
+internal fun TextFieldComposable(
```

Comment:
Would it make sense to keep this composable function named `TextField`? On one hand, it's our own wrapper around `TextField` to allow using `TextFieldElement` to configure the widget which will work just fine. On the other, it might make things a bit more confusing as someone might use the plain `TextField` and end up having to redo a bunch of the processing already handled in this function (state, colour and focus processing).